### ISSUE TITLE
[BUG] Installing Salt for development docs out of date

### ISSUE THREAD

**Issue by SammyTbeile** [saltstack/salt#58516]
**Description**
The documentation for Installing Salt for development feels like it's out of date https://docs.saltstack.com/en/latest/topics/development/hacking.html . One example is that it seems to indicate that Python2 should be used:
`On Arch Linux, where Python 3 is the default installation of Python, use the virtualenv2 command instead of virtualenv.`

**Comment by sagetherage** [comment]
Thank you for submitting this issue, due to demands we likely will not get to this issue in the Magnesium release, so setting it for Aluminium.

**Comment by ScriptAutomate** [comment]
As an added note, that docs page has a section that will be updated to either include or point to a dedicated section around contributing to Salt documentation.

I'm referring to this section:
- https://docs.saltstack.com/en/latest/topics/development/hacking.html#editing-and-previewing-the-documentation

We are having an upcoming docs clinic where we are working to improve/update docs workflow for contributors, with this WIP PR being discussed in a different repo (via the Docs Working Group) before we eventually port to the official `salt` docs here. WIP docs contributor guide:
- https://github.com/saltstack/docs-hub/pull/68

**Comment by ScriptAutomate** [comment]
For generating the docs, with `nox` already installed:

```
# Test building out HTML / Sphinx
nox -e 'docs-html(compress=False)'
```

**Comment by ScriptAutomate** [comment]
Related PR in progress: https://github.com/saltstack/salt/pull/57645

**Comment by sagetherage** [comment]
this is related to the PR above that was merged - is there more outstanding work for `Aluminium` @ScriptAutomate ? or can we plan that work for another release?

**Comment by ScriptAutomate** [comment]
This issue is resolved by #57645 :)
